system-tests/lib/fixtures.ts

```diff
@@ -49,12 +53,32 @@ export function scaffold () {
 
 /**
  * Given a project name, copy the project's test files to the temp dir.
+ * Returns the scaffolded directory
  */
-export async function scaffoldProject (project: string): Promise<void> {
+export async function scaffoldProject (project: string): Promise<string> {
   const to = _path.join(cyTmpDir, project)
-  const from = _path.join(projects, project)
+  const from = projectFixturePath(project)
 
   await fs.copy(from, to)
+
+  try {
+    const packageJson = require(`${to}/package.json`)
+    const fixtureDir = packageJson.projectFixtureDirectory
+
+    if (fixtureDir) {
+      if (!projectFixtureDirs.includes(fixtureDir)) {
+        throw new Error(`Invalid project fixture directory: ${fixtureDir}, expected one of ${projectFixtureDirs}`)
+      }
+
+      await fs.copy(_path.join(projectFixtures, fixtureDir), to)
```

Comment:
Does this overwrite or merge with existing directories in the project dir? Merging might help keep the fixtures from similarly ballooning in number by allowing additive tweaks at the project level. Or it might just make code tracing more complicated 🤷‍♂️

Reply:
No, it just writes the files in the chosen `project-fixtures` within a given individual directory after scaffolding. The intent is that the project we're merging into will only have a `package.json` / `yarn.lock` file, and we're copying the fixture project we want to test against different dependency versions into it.

See the blurb about it here: https://github.com/cypress-io/cypress/blob/7fd7a6aa2e85776e81d10da81d1b966c121dc7fb/system-tests/README.md#reusable-project-fixtures

Reply:
That is to say, the main intent, at least to start, is that the additive tweaks are limited to different package versions. In the future we can get fancier with it as the use cases arise.